src/hooks/handleQueryInUrl.ts

```diff
@@ -1,27 +1,37 @@
-import {
-  usePageContext,
-  usePanelContext,
-  useQueriesCurrentForPage,
-} from "@/hooks";
+import { getPageForQuery } from "@/helpers";
+import { usePageContext, usePanelContext, useQueries } from "@/hooks";
 import type { OracleQueryUI } from "@/types";
-import filter from "lodash/filter";
+import { find } from "lodash";
 import { useRouter } from "next/router";
 import { useEffect } from "react";
 
 export function useHandleQueryInUrl() {
   const router = useRouter();
   const { openPanel, panelOpen } = usePanelContext();
+  const { all: queries } = useQueries();
   const { page } = usePageContext();
-  const queries = useQueriesCurrentForPage();
 
   useEffect(() => {
     const hasQueryUrl = Object.keys(router.query).length > 0;
-    const queryResults: OracleQueryUI[] = filter<OracleQueryUI>(
-      queries,
-      router.query
-    );
-    if (hasQueryUrl && queryResults.length === 1 && !panelOpen) {
-      openPanel(queryResults[0], page, false);
+    if (!hasQueryUrl) return;
+
+    const query = find<OracleQueryUI>(queries, router.query);
```

Comment:
the reason filter was used here is because its possible to search on parameters which might match multiple queries. if thats the case, im not sure we should pick one for the user. `find` will just return the first one found. not a huge deal and we can debate for future pr